vars/getCompareBenchmarkIds.groovy

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+import groovy.json.JsonSlurper
+
+Map<String, String> call(Map args = [:]) {
+    def isNullOrEmpty = { str ->
+        str == null || (str instanceof String && str.trim().isEmpty())
+    }
+    String baselineId = getBaselineTestExecutionId(args.baselineClusterConfig, args.distributionVersion, args.workload)
+    String contenderId = getContenderTestExecutionId(args.pullRequestNumber)
+    return ['baseline': baselineId, 'contender': contenderId]
+}
+
+String getBaselineTestExecutionId(baselineClusterConfig, distributionVersion, workload) {
+    withCredentials([string(credentialsId: 'benchmark-metrics-datastore-user', variable: 'DATASTORE_USER'),
+                     string(credentialsId: 'benchmark-metrics-datastore-password', variable: 'DATASTORE_PASSWORD'),
+                     string(credentialsId: 'benchmark-metrics-datastore-endpoint', variable: 'DATASTORE_ENDPOINT')]) {
+        def curlCommand = """
+              curl -X POST "https://${DATASTORE_ENDPOINT}/benchmark-results-*/_search" -ku ${DATASTORE_USER}:${DATASTORE_PASSWORD} -H 'Content-Type: application/json' -d '{
+              "size": 1,
+              "query": {
+                "bool": {
+                  "must": [
+                    {
+                      "term": {
+                        "user-tags.cluster-config": \"${baselineClusterConfig}\"
+                      }
+                    },
+                    {
+                      "term": {
+                        "workload": \"${workload}\"
+                      }
+                    },
+                    {
+                      "term": {
+                        "distribution-version": \"${distributionVersion}\"
+                      }
+                    },
+                    {
+                      "range": {
+                        "test-execution-timestamp": {
+                          "gte": "now-5d/d",
+                          "lte": "now/d"
+                        }
+                      }
+                    }
+                  ]
+                }
+              },
+              "sort": [
+                {
+                  "test-execution-timestamp": {
+                    "order": "desc"
+                  }
+                }
+              ],
+              "_source": ["test-execution-id"]
+            }'
+        """
+
+        println("The curl command is:\n${curlCommand}")
```

Comment:
Shall we remove this since it has risk of exposing the credentials?

Reply:
Removed, but don't think there is any risk of exposing credentials as jenkins masks all the values since it is used in the scope of `withCredentials` method. 
Have tested and verified this already.